pkg/resource/table/hooks.go

```diff
@@ -283,12 +335,16 @@ func (rm *resourceManager) setResourceAdditionalFields(
 ) (err error) {
 	rlog := ackrtlog.FromContext(ctx)
 	exit := rlog.Trace("rm.setResourceAdditionalFields")
-	defer exit(err)
+	defer func(err error) { exit(err) }(err)
 
 	ko.Spec.Tags, err = rm.getResourceTagsPagesWithContext(ctx, string(*ko.Status.ACKResourceMetadata.ARN))
 	if err != nil {
 		return err
 	}
+	ko.Spec.TimeToLive, err = rm.getResourceTTLWithContext(ctx, ko.Spec.TableName)
```

Comment:
The reason I don't care for this style of setting `ko.Spec` fields is because if `err` is not nil, we will override `ko.Spec.TimeToLive` to whatever happened to be returned from `getResourceTTLWithContext`, which will be `nil`. We actually do *not* want to do this, but rather only set the `ko.Spec` value if we do not get an error returned from `getResourceTTLWithContext`. So, this is what we should be doing:

```go
var ttl *v1alpha1.TimeToLiveSpecification
if ttl, err := rm.getResourceTTLWithContext(ctx, ko.Spec.TableName); err != nil {
    return err
} else {
    ko.Spec.TimeToLive = ttl
}
```

(same goes for the tags above on line 340...)